Sources/AppcuesKit/Data/Analytics/SessionMonitor.swift

```diff
@@ -56,42 +54,32 @@ internal class SessionMonitor: SessionMonitoring {
     }
 
     // called on (A) sdk init (B) user identity change
-    func start() {
+    func start() -> Bool {
         // if there is no user identified, we do not have a valid session
-        guard !storage.userID.isEmpty else { return }
-
+        guard !storage.userID.isEmpty else { return false }
         appcues?.sessionID = UUID()
-        publisher.track(SessionEvents.sessionStarted, properties: nil, interactive: true)
+        updateLastActivity()
+        return true
+    }
+
+    func updateLastActivity() {
+        guard appcues?.sessionID != nil else { return }
+
+        lastActivityAt = Date()
     }
 
     // called on reset(), user sign-out
     func reset() {
-        // this is interactive: true since a reset should flush to network immediately (with previous user ID)
-        // and the next session start will be sent in a new request, with the new user ID
-        publisher.track(SessionEvents.sessionReset, properties: nil, interactive: true)
-        appcues?.sessionID = nil
-    }
+        // ensure any pending in-memory analytics get processed asap
+        tracker.flush()
```

Comment:
yep thats correct - I think technically it might work still if things just went out on the queue, but wanted to be safe since the user ID and session ID are getting cleared out right after this.